packages/portal/src/Portal.tsx

```diff
@@ -1,80 +1,67 @@
-import { Component } from 'react';
+import { useEffect, useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 import { createPortal } from 'react-dom';
 
-interface PortalContainer {
-  el: Element;
-  remove: () => Element;
-}
-
-interface PortalProps {
+type PortalProps = {
   children?: React.ReactNode;
-  container?: HTMLElement;
-}
+} & (
+  | { container: HTMLElement; className?: never }
+  | { container?: never; className?: string }
```

Comment:
Is there a reason `container` and `className` need to be mutually exclusive. We're expecting an HTML element, not a React node here, so we should be able to append the className to the elements classList without making any more assumptions than we already are.

If we do go this route, this looks like it may be a perfect case for the `Either` TS helper that we added previously to lib (`packages/lib/index.ts:48` I believe).

Reply:
I usually lean towards less permissive interfaces so that we can change our minds later without creating a breaking change. Also, I figure since the caller has a reference to the HTML element they can just add the class name themselves.

I saw `Either`, but it wouldn't have the same behavior because:
- it doesn't enforce mutual exclusivity (which we get from using `never`)
- doesn't allow all properties to be optional (because it uses `]-?:` in the mapped type)

Reply:
Both points are fair enough! Also, whoops – for some reason, I was thinking of Either as enforcing mutual exclusivity. I think this could be a great helper to add to lib if you'd like! I think we'd definitely use this in the future.